### PR TITLE
Add support for realm-specific configuration overrides

### PR DESCRIPTION
**Pull request by collado-mike**
# Description

The `PolarisConfigurationStore` allows for server-level configuration parameters, but doesn't allow for realm-specific configuration overrides. In a multi-tenant installation, realm-specific configuration should be expected and supported.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Adds unit test for `DefaultConfigurationStore`

- [X] DefaultConfigurationStoreTest

**Test Configuration**:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

Please delete options that are not relevant.

- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] If adding new functionality, I have discussed my implementation with the community using the linked GitHub issue

### CODE REVIEW

File: polaris-service/src/main/java/org/apache/polaris/service/config/DefaultConfigurationStore.java [hunk 1]
```diff
@@ -21,18 +21,30 @@
...
 import org.apache.polaris.core.PolarisCallContext;
 import org.apache.polaris.core.PolarisConfigurationStore;
+import org.apache.polaris.core.context.CallContext;
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 public class DefaultConfigurationStore implements PolarisConfigurationStore {
-  private final Map<String, Object> properties;
+  private final Map<String, Object> config;
+  private final Map<String, Map<String, Object>> realmConfig;
```
Comment (eric-maynard) [inline polaris-service/src/main/java/org/apache/polaris/service/config/DefaultConfigurationStore.java:30]: nit: Maybe a name like globalConfig or fallbackConfig would be more clear

File: polaris-service/src/main/java/org/apache/polaris/service/config/PolarisApplicationConfig.java [hunk 2]
```diff
@@ -170,7 +169,12 @@ public TaskHandlerConfiguration getTaskHandler() {
 
   @JsonProperty("featureConfiguration")
   public void setFeatureConfiguration(Map<String, Object> featureConfiguration) {
-    this.configurationStore = new DefaultConfigurationStore(featureConfiguration);
+    this.featureConfiguration = featureConfiguration;
+  }
+
+  @JsonProperty("realmFeatureConfiguration")
```
Comment (eric-maynard) [inline polaris-service/src/main/java/org/apache/polaris/service/config/PolarisApplicationConfig.java:175]: What does the config file look like in this case? It seems to be something like...

```
featureConfiguration:
  SOME_FEATURE: true

realmFeatureConfiguration:
  realm_1:
    SOME_FEATURE: true
  realm_2:
    SOME_FEATURE: false
```

Can we do any validation that these realms actually exist?

File: polaris-service/src/test/java/org/apache/polaris/service/config/DefaultConfigurationStoreTest.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Map;
+import org.apache.polaris.core.PolarisCallContext;
+import org.apache.polaris.core.PolarisDefaultDiagServiceImpl;
+import org.apache.polaris.service.persistence.InMemoryPolarisMetaStoreManagerFactory;
+import org.junit.jupiter.api.Test;
+
+class DefaultConfigurationStoreTest {
```
Comment (eric-maynard) [inline polaris-service/src/test/java/org/apache/polaris/service/config/DefaultConfigurationStoreTest.java:29]: Other tests are `public`

**Review by eric-maynard — APPROVED** [review]
LGTM overall, left a few small comments

File: polaris-service/src/main/java/org/apache/polaris/service/config/PolarisApplicationConfig.java (hunk 2 above)
Comment (collado-mike) [reply polaris-service/src/main/java/org/apache/polaris/service/config/PolarisApplicationConfig.java:175]: yeah, that's the proposed configuration structure in the yaml file. No, we can't validate that the realms exist currently - there is no agreed upon interface for storing/retrieving realms and the only persistence API doesn't know anything about realms.

File: polaris-service/src/test/java/org/apache/polaris/service/config/DefaultConfigurationStoreTest.java (hunk 3 above)
Comment (collado-mike) [reply polaris-service/src/test/java/org/apache/polaris/service/config/DefaultConfigurationStoreTest.java:29]: Blame intellij :)

**Review by eric-maynard — APPROVED** [review]

**Review by eric-maynard — APPROVED** [review]

**Pull request merged**